crates/api/src/local_user/resend_verification_email.rs

```diff
@@ -0,0 +1,32 @@
+use actix_web::web::{Data, Json};
+use lemmy_api_common::{
+  context::LemmyContext,
+  person::ResendVerificationEmail,
+  utils::send_verification_email,
+  SuccessResponse,
+};
+use lemmy_db_views::structs::{LocalUserView, SiteView};
+use lemmy_utils::error::LemmyResult;
+
+pub async fn resend_verification_email(
+  data: Json<ResendVerificationEmail>,
+  context: Data<LemmyContext>,
+) -> LemmyResult<Json<SuccessResponse>> {
+  let site_view = SiteView::read_local(&mut context.pool()).await?;
+  let email = data.email.to_string();
+
+  // Fetch that email
+  let local_user_view = LocalUserView::find_by_email(&mut context.pool(), &email).await?;
+
```

Comment:
You need to check that the account isnt verified yet.

Reply:
I noticed there are two functions for sending verification email, one of them already includes this check:
```
crates/api_common/src/utils.rs
463:pub async fn send_verification_email(

crates/api_crud/src/user/create.rs
478:async fn send_verification_email_if_required(
```

Would be good to merge them into one.

Reply:
K updated. I'm using the `send_verification_email_if_required` function, which returns `Option<bool>` (and doesn't send under a few cases). But I'm ignoring that result, because if we add an error, it might be another one of email enumeration things.

Reply:
You should also be able to get rid of `send_verification_email()` and merge that code directly into `send_verification_email_if_required()`.

Reply:
I checked and it can't, because it needs to skip those checks (and use a custom email), when saving user settings.

Reply:
Alright